components/organisms/CollaborationRequestWrapper/collaboration-requests-wrapper.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState } from "react";
+import { format, isToday, isYesterday, formatDistanceToNowStrict } from "date-fns";
+import CollaborationCard from "components/molecules/CollaborationCard/collaboration-card";
+import { useUserCollaborations } from "lib/hooks/useUserCollaborations";
+import DashContainer from "components/atoms/DashedContainer/DashContainer";
+
+function formatPostDate(date: Date) {
+  if (isToday(date)) {
+    return "Today";
+  } else if (isYesterday(date)) {
+    return "Yesterday";
+  } else {
+    return formatDistanceToNowStrict(date, { addSuffix: true });
+  }
+}
+
+const CollaborationRequestsWrapper = () => {
+  const { data, updateCollaborationStatus, deleteCollaborationRequest } = useUserCollaborations();
+  let currentDate: string;
+  console.log(data);
+
+  const getDateGroupHeader = (date: Date, index: number) => {
+    const dateString = date.toLocaleDateString();
+    if (dateString !== currentDate) {
+      currentDate = dateString;
+
+      return (
+        <p className={`mt-4 text-light-slate-10 md:pt-4 md:mt-8 ${index !== 0 && " border-t"}`}>
+          {formatPostDate(new Date(date))}
+        </p>
+      );
+    }
+    return;
+  };
+
+  return (
+    <div>
+      {data && data.length > 0 ? (
+        data
```

Comment:
Create a new array if you're going to sort in place
```suggestion
        data.slice()
```